cosmos/x/evm/keeper/processor.go

```diff
@@ -22,45 +22,51 @@ package keeper
 
 import (
 	"context"
+	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"pkg.berachain.dev/polaris/eth/core"
 	coretypes "pkg.berachain.dev/polaris/eth/core/types"
 )
 
 // ProcessTransaction is called during the DeliverTx processing of the ABCI lifecycle.
-func (k *Keeper) ProcessTransaction(ctx context.Context, tx *coretypes.Transaction) (*core.ExecutionResult, error) {
+func (k *Keeper) ProcessTransaction(ctx context.Context, tx *coretypes.Transaction) (*coretypes.Receipt, error) {
 	sCtx := sdk.UnwrapSDKContext(ctx)
+	gasMeter := sCtx.GasMeter()
 	// We zero-out the gas meter prior to evm execution in order to ensure that the receipt output
 	// from the EVM is correct. In the future, we will revisit this to allow gas metering for more
 	// complex operations prior to entering the EVM.
-	sCtx.GasMeter().RefundGas(sCtx.GasMeter().GasConsumed(),
+	gasMeter.RefundGas(gasMeter.GasConsumed(),
 		"reset gas meter prior to ethereum state transition")
 
 	// Process the transaction and return the EVM's execution result.
-	execResult, err := k.miner.ProcessTransaction(ctx, tx)
+	receipt, err := k.miner.ProcessTransaction(ctx, tx)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
I think this case should also log the error

Reply:
yeah sure